### ISSUE TITLE
GTA relative transformation

### ISSUE THREAD

**Issue by MichaelJacob914** [autonomousvision/gta#3]
Hi, very nice codebase, it is very well-designed. I am wondering if the implementation in this code aligns with the equation presented in the paper, or if I am misunderstanding it. 

<img width="862" height="120" alt="Image" src="https://github.com/user-attachments/assets/b47437dd-dd2b-43ce-9e4e-52f8cabc474b" />

Equation 5 describes transforming K and V into the frame of Q. From my understanding of the code, however, the opposite will be done: 
                fn_so3(D_q[i], q_so3[..., end_dim-dim:end_dim]))
                fn_so3(D_k[i], k_so3[..., end_dim-dim:end_dim]))
                fn_so3(D_k[i], v_so3[..., end_dim-dim:end_dim])
                          if v_transform else v_so3[..., end_dim-dim:end_dim])
The forward transform is applied to all 3, and then when D_q is transposed during attention it will yield 
$$Q_{i}p_{g_{i}}^{-1}p_{g_{j}}K_{j}$$ 
This seems to match with the end,  since O is computed as 
O = fn_so3(inv_D_q[i].detach(), out_so3[..., end_dim-dim:end_dim]))
which corresponds to $p_{g_{i}}^{-1}$ times the summation as opposed to $p_{g_{i}}$, which I think would switch the order of transformations in Eq(5). 

Any clarification is appreciated.

**Comment by MichaelJacob914** [comment]
Thank you very much!
